File: JadiNih/app/src/main/java/com/example/jadinih/MainActivity.java
package com.example.jadinih;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

public class MainActivity extends AppCompatActivity {

    TextView nilaiSensor;
    TextView jam;
    TextView menit;
    TextView detik;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        nilaiSensor = (TextView) findViewById(R.id.nilaiSensor);
        jam = (TextView) findViewById(R.id.jam);
        menit = (TextView) findViewById(R.id.menit);
        detik = (TextView) findViewById(R.id.detik);
        DatabaseReference koneksi = FirebaseDatabase.getInstance().getReference();

        koneksi.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot snapshot) {
                String Nilai = snapshot.child("nilai").getValue().toString();
                nilaiSensor.setText(Nilai);
                String Jam = snapshot.child("jam").getValue().toString();
                jam.setText(Jam);
                String Menit = snapshot.child("menit").getValue().toString();
                menit.setText(Menit);
                String Detik = snapshot.child("detik").getValue().toString();
                detik.setText(Detik);
            }

            @Override
            public void onCancelled(@NonNull DatabaseError error) {

            }
        });

    }
}

File: MotitoringAquarium/app/src/main/java/com/example/motitoringaquarium/MainActivity.kt
package com.example.motitoringaquarium

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        getDataFromFirebase()

    }

    private fun getDataFromFirebase() {

        val jam: TextView = findViewById(R.id.jam) as TextView
        val menit: TextView = findViewById(R.id.menit) as TextView
        val detik: TextView = findViewById(R.id.detik) as TextView
        val nilai: TextView = findViewById(R.id.nilai) as TextView

        FirebaseDatabase.getInstance()
            .getReference("data")
            .addValueEventListener(object : ValueEventListener {
                override fun onDataChange(snapshot: DataSnapshot) {
                    val Jam = snapshot.child("jam").getValue().toString()
                    val Menit = snapshot.child("menit").getValue().toString()
                    val Detik = snapshot.child("detik").getValue().toString()
                    val Nilai = snapshot.child("tds").getValue()

                    jam.setText(Jam)
                    menit.setText(Menit)
                    detik.setText(Detik)
                    nilai.setText(Nilai.toString())
                }

                override fun onCancelled(error: DatabaseError) {
                    TODO("Not yet implemented")
                }
            })


    }
}


